### ISSUE TITLE
No 1-dimensional max pool?

### ISSUE THREAD

**Issue by aetilley** [tensorflow/tensorflow#9442]
It seems that there is no 1-dimensional max pool.  

I'm using one for my own projects.  Should I submit a pull request?

**Comment by girving** [comment]
Yep, a pull request would be welcome.  Please do it in pure Python, though: just a thin `tf.nn.max_pool1d` wrapper calling down to `tf.nn.max_pool`.

**Comment by aetilley** [comment]
Will do.

**Comment by girving** [comment]
Hmm, there's already a `max_pooling1d` in layers.  If you make a non-layer `max_pool1d`, it might be good to share code with the layer.  That is, you'd pull the layer code out into a separate function, and then make the layer code use the function.

**Comment by aetilley** [comment]
@girving 
That's interesting.   I submitted a pull request for the first change.  I wasn't sure if it was a good idea to touch the layers bit, but feel free to edit.

I also noticed this on line 784:

    pooling_ops = {("MAX", 1): max_pool,                                                                           
                   ("MAX", 2): max_pool,                                                                           
                   ("MAX", 3): max_pool3d,  # pylint: disable=undefined-variable                                   
                   ("AVG", 1): avg_pool,                                                                           
                   ("AVG", 2): avg_pool,                                                                           
                   ("AVG", 3): avg_pool3d,  # pylint: disable=undefined-variable                                   
                  }

This is suggesting that max_pool functions as for dim 1, but I don't think this will work.

**Comment by girving** [comment]
Hmm, actually that looks like it works great.  Note that `expand_dims` and `squeeze` that happen if `num_spatial_dims == 1`.  I don't think your PR will be necessary.

**Comment by girving** [comment]
Can you verify that it works for you?

**Comment by aetilley** [comment]
@girving 

That's fine.  But note that there's also a generic N-dim convolution function, as well as specialized 1,2,3 dim versions.

It's up to you.

**Comment by girving** [comment]
The generic form is newer than the specialized versions, and I think it's nicer.

**Comment by aetilley** [comment]
@girving 

Got it.  Thanks.

**Comment by Leechen2014** [comment]
there https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/layers/pooling.py
given a function : pooling1d()
